### PR TITLE
IW-1561 add at-mention email controller

### PR DESCRIPTION
**Pull request by jsutterfield**
Ticket: https://wikia-inc.atlassian.net/browse/IW-1561

This PR adds a new Discussion's email for at-mentions. It will be called by the notification service whenever a user is at-mentioned in a thread or post.

Related PR: https://github.com/Wikia/pandora/pull/6596

/cc @Wikia/iwing

### CODE REVIEW

File: extensions/wikia/Email/Controller/DiscussionController.class.php [hunk 1]
```diff
@@ -77,7 +77,7 @@ protected function getContentFooterMessages() {
 	}
 
 	private function getDiscussionsLink() {
-		return $this->wiki->city_url . 'd';
+		return $this->wiki->city_url . 'f';
```
Comment (rybmat) [inline extensions/wikia/Email/Controller/DiscussionController.class.php:80]: does city_url contain trailing '/'?

File: extensions/wikia/Email/Controller/DiscussionController.class.php [hunk 2]
```diff
@@ -290,5 +290,69 @@ public function assertCanEmail() {
 			throw new Check( 'User is not subscribed to Discussions Upvote emails.' );
 		}
 	}
+}
 
+class DiscussionAtMentionController extends DiscussionController {
```
Comment (rybmat) [inline extensions/wikia/Email/Controller/DiscussionController.class.php:295]: maybe we should put that specific classes in separate files?

**Review by rybmat — APPROVED** [review]

**Review by mszabo-wikia — APPROVED** [review]

File: extensions/wikia/Email/Controller/DiscussionController.class.php (hunk 1 above)
Comment (jsutterfield) [reply extensions/wikia/Email/Controller/DiscussionController.class.php:80]: Yep, looks like it does:
```
dev-james:maintenance (IW-1561) james$ SERVER_ID=3035 php eval.php
> print_r( \WikiFactory::getWikiByID( 3035 )->city_url );
http://fallout.wikia.com/
```

File: extensions/wikia/Email/Controller/DiscussionController.class.php (hunk 2 above)
Comment (jsutterfield) [reply extensions/wikia/Email/Controller/DiscussionController.class.php:295]: Eh, I kind of like this as is. It's the convention used through the email controllers (so it's consistent), and I think it makes it easier to see what methods and member variables these sub-classed controllers inherit. Unless you have strong feelings about updating that, I think I'll leave like this :)

**Pull request merged**